### PR TITLE
Add first draft of abstraction for the ts3 layer.

### PR DESCRIPTION
**Pull request by Xyaren**
This are the first steps to encapsulate the ts3 connection into a more developer friendly repository.
Tests seem to be very hard, because we can not inspect the lambda expression, that is handed to the TS3Connection.

### CODE REVIEW

File: TS3Bot.py [hunk 1]
```diff
@@ -55,168 +43,16 @@ def __init__(self, db_name):
...
-            log.info("Renamed self to '%s'.", nickname)
-        else:
-            self.ts3exec(lambda tc: tc.exec_("clientupdate", client_nickname=nickname))
-            log.info("Forcefully renamed self to '%s'.", nickname)
-        self._bot_nickname = nickname
-        return self._bot_nickname
-
-
 class Bot:
+    _ts_connection: ThreadsafeTSConnection
```
Comment (ogrady) [inline TS3Bot.py:47]: this creates a static variable. Python implicitly declares attributes by assigning them from anywhere, preferably the constructor.

File: TS3Bot.py [hunk 2]
```diff
@@ -55,168 +43,16 @@ def __init__(self, db_name):
...
+    _ts_repository: TS3Repository
```
Comment (ogrady) [inline TS3Bot.py:48]: same as above.

File: TS3Bot.py (hunk 1 above)
Comment (Xyaren) [reply TS3Bot.py:47]: I was trying to add type hinting, how whould you do it ?

File: TS3Bot.py (hunk 2 above)
Comment (Xyaren) [reply TS3Bot.py]: fixed

File: TS3Bot.py (hunk 1 above)
Comment (Xyaren) [reply TS3Bot.py]: fixed.

**Pull request merged**